README.md

```diff
@@ -428,9 +501,63 @@ hook.remove(name, hookMethod)
 Example
 
 ```js
-hook.remove('save', validateRecord)
+hookCollection.remove('save', validateRecord)
+```
+
+## TypeScript
+
+This library contains type definitions for TypeScript. When you use TypeScript we highly recommend using the `Hook.Singular` constructor for your hooks as this allows you to pass along type information for the options object. For example:
+
+```ts
+
+import {Hook} from 'before-after-hook'
+
+interface Foo {
+  bar: string
+  num: number;
+}
+
+const hook = new Hook.Singular<Foo>();
+
+hook.before(function (foo) {
+
+  // typescript will complain about the following mutation attempts
+  foo.hello = 'world'
+  foo.bar = 123
+
+  // yet this is valid
+  foo.bar = 'other-string'
+  foo.num = 123
+})
+
+const foo = hook({bar: 'random-string'}, function(foo) {
+  // handle `foo`
+  foo.bar = 'another-string'
+})
+
+// foo outputs
+{
+  bar: 'another-string',
+  num: 123
+}
 ```
 
+An alternative import:
+```ts
+import {Singular, Collection} from 'before-after-hook'
+
+const hook = new Singular<{foo: string}>();
+const hookCollection = new Collection();
+```
+
+## Upgrading to 1.4
+
+Since version 1.4 the `Hook` constructor has been deprecated in favor of becoming a singleton in the next major release. 
```

Comment:
I think `singleton` is wrong here, singleton is a pattern where you export an instance of something, and each file that requires it receives the same instance. Maybe replace it the above sentence with something like

```suggestion
Since version 1.4 the `Hook` constructor has been deprecated in favor of returning `Hook.Singular` in the next major release. 
```

This is not a blocker, we can fix this after the new release